root/static/scripts/edit/URLCleanup.js

```diff
@@ -1342,6 +1342,16 @@ const CLEANUPS: CleanupEntries = {
       return {result: false, target: ERROR_TARGETS.URL};
     },
   },
+  'bugs': {
+    match: [new RegExp('^(https?://)?(music|m)\\.bugs\\.co\\.kr/album/', 'i')],
+    restrict: [multiple(LINK_TYPES.downloadpurchase, LINK_TYPES.streamingpaid), LINK_TYPES.streamingpaid],
+    select: function (url, sourceType) {
+      return [LINK_TYPES.downloadpurchase.release, LINK_TYPES.streamingpaid.release];
+    },
+    clean: function (url) {
+      return url.replace(/^(?:https?:\/\/)?(?:music|m)\.bugs\.co\.kr\/album\/(\d+)(?:[\/?#].*)?$/, 'https://music.bugs.co.kr/album/$1');
+    },
```

Comment:
Cleaning album URLs is fine. Would it be worth cleaning artist and track URLs too?

Reply:
I don't think so, I haven't seen them linked yet to artists anywhere.
I could of course do it if we want it.

Reply:
There currently are 408 Bugs! albums, 59 Bugs! artists, 4 Bugs! tracks, 2 Bugs! movies, and 2 Bugs! search pages linked into the MusicBrainz database. Albums are definitely the main links, but artists are not negligible either. Tracks and movies are probably not worth the effort on their own. Search results are definitely to be avoided. That’s why URL handlers usually supports a website as a whole: to help editors with linking appropriate clean URLs to the corresponding entity type, showing explanatory error message otherwise; See `noLinkToSearchMsg` for example.

If you still decide to support Bugs! albums only, make it clear by using `bugs_album` instead of `bugs`.

Reply:
I see. If I wanted to support both artists and albums, I would just need to add another matcher for artists, right?
And maybe another matcher for search results that throws an error?

Reply:
If you wanted to support more than albums, I would recommend you to use only one matcher (I mean `CLEANUPS` entry) for all Bugs! URLs. It is actually less effort than making separate matchers and makes it easier to maintain in the long run.

Reply:
Right, I actually meant multiple RegExp entries in the match field, not multiple CLEANUPS entries. Sorry for the confusion.

Reply:
Multiple RegExp entries if you want just to handle some subpaths, or only one RegExp entry if you want to handle all URLs for that website.

Reply:
Here is a rough list of current Bugs! URLs in the MBDB: https://gist.github.com/yvanzo/fb5d13c31776eed64dc30e845d0388c7

Reply:
Thanks, that was super helpful! I extended the Bugs! entry to clean and validate all other URLs as well. Let me know if I did everything correctly, then I'll attempt the same for Melon.